src/op/compute_op.cc

```diff
@@ -457,11 +457,11 @@ ComputeLoopNest ComputeLoopNest::make(
       ret.init_vmap[iv] = ret.main_vmap.at(iv);
     }
     ret.num_common_loop = begin_loop;
-    // skip loops that does not relates to axis.
+    // skip loops that are related to reduction axes.
     std::unordered_set<IterVar> skip_iter;
     for (auto kv : update_state) {
       int flag = kv.second;
-      if ((flag & 1) == 0) skip_iter.insert(kv.first);
+      if ((flag & 2) != 0) skip_iter.insert(kv.first);
     }
```

Comment:
why not `flag == 2`?

Reply:
Good point, this actually should be `flag == 2`. Added a test case for this.